data_structures/route_parameters.cpp

```diff
@@ -117,6 +117,15 @@ void RouteParameters::addTimestamp(const unsigned timestamp)
     }
 }
 
+void RouteParameters::addBearing(const int bearing)
+{
+    bearings.resize(coordinates.size());
+    if (!bearings.empty())
+    {
+        bearings.back() = bearing;
+    }
+}
```

Comment:
Let me see if I understand this correctly: this is the handler for the `b=` spirit grammar, called with ints. It implicitly relies on interleaving coordinates and bearings. In the case of multiple bearings such as `b=1&b=2&b=3` only the last `b=3` is stored.